virtual_ecosystem/models/hydrology/above_ground.py

```diff
@@ -15,6 +14,176 @@
+    density_air_kg: NDArray[np.float32],
+    specific_heat_air: NDArray[np.float32],
+    aerodynamic_resistance: NDArray[np.float32],
+    stomatal_resistance: NDArray[np.float32],
+    latent_heat_vapourisation: NDArray[np.float32],
+    psychrometric_constant: NDArray[np.float32],
+    saturated_pressure_slope_parameters: tuple[float, float, float, float],
+):
+    """Calculate canopy potential evaporation rate using Penman-Monteith equation.
+
+    Args:
+        net_radiation: Net radiation at leaf surface, [W m-2]
+        vapour_pressure_deficit: Vapour pressure deficit, [kPa]
+        air_temperature: Air temperature, [C]
+        density_air_kg: Air density, [kg m-3]
+        specific_heat_air: Specific heat of air, [J kg-1 K-1]
+        aerodynamic_resistance: Aerodynamic resistance in canopy, [s m-1]
+        stomatal_resistance: Stomatal resistance, [s m-1]
+        latent_heat_vapourisation: Latent heat of vapourisation, [J kg-1]
+        psychrometric_constant: Psychrometric constant, [kPa K-1]
+        saturated_pressure_slope_parameters: List of parameters to calculate
+            the slope of the saturated vapour pressure curve
+
+    Returns:
+        potential evaporation rate, [kg m-2 s-1]
+    """
+
+    # Slope of saturation vapour pressure curve (kPa/K)
+    delta = calculate_slope_of_saturated_pressure_curve(
+        temperature=air_temperature,
+        saturated_pressure_slope_parameters=saturated_pressure_slope_parameters,
+    )
+
+    # Penman-Monteith equation
+    potential_evaporation = (
+        delta * net_radiation
+        + density_air_kg
+        * specific_heat_air
+        * (vapour_pressure_deficit / aerodynamic_resistance)
+    ) / (
+        latent_heat_vapourisation
+        * (
+            delta
+            + psychrometric_constant
+            * (1 + stomatal_resistance / aerodynamic_resistance)
+        )
+    )
+
+    return potential_evaporation
+
+
+def calculate_canopy_evaporation(
+    leaf_area_index: NDArray[np.float32],
+    interception: NDArray[np.float32],
+    net_radiation: NDArray[np.float32],
+    vapour_pressure_deficit: NDArray[np.float32],
+    air_temperature: NDArray[np.float32],
+    density_air_kg: NDArray[np.float32],
+    specific_heat_air: NDArray[np.float32],
+    aerodynamic_resistance: NDArray[np.float32],
+    stomatal_resistance: NDArray[np.float32],
+    latent_heat_vapourisation: NDArray[np.float32],
+    psychrometric_constant: NDArray[np.float32],
+    saturated_pressure_slope_parameters: tuple[float, float, float, float],
+    time_interval: float,
+    intercept_residence_time: float,
+    extinction_coefficient_global_radiation: float,
+) -> dict[str, NDArray[np.float32]]:
+    r"""Calculate evaporation of intercepted water from the canopy, [mm].
+
+    This function calculates evaporation of intercepted water from the canopy assuming
+    the potential evaporation rate from an open water surface :math:`EW_{0}` [mm] (
```

Comment:
make clear that the overall method is from LISFLOOD. potential evaporation is based on Penman Monteith; LISFLOOD uses data from external sources and does not calculate this explicitly